src/logic/calculate.js

```diff
@@ -0,0 +1,24 @@
+import operate from './operate';
+
+const calculate = ((data, buttonName) => {
+  let { total, next, operation } = data;
+
+  if (buttonName === '+/-') {
+    total = -total;
+    next = -next;
+    operation = buttonName;
+    return total;
+  }
+  if (buttonName === 'AC') {
+    total = 0;
+    return total;
+  }
+  if (buttonName === '.' && next !== false) {
+    total = `${total}.${next}`;
+    return total;
+  }
```

Comment:
Let's handle cases where:
- `next` and `total` is `undefined`, "0." should be returned
- when `next` is defined, `.` should be appended to it